Articles/Blog/HelloHeatEquation.md

```diff
@@ -0,0 +1,226 @@
+* Heat only moves horizontally through the bar
+* There are no external heating/cooling sources
+
+<div style="text-align:center"><img width="75%" src="https://github.com/betterscientificsoftware/images/blob/master/Blog_0719_HeatEqnBar.png" /></div>
+
+To model the temperature distribution of the rod, we can use the 1-dimensional Heat Equation given by the formula:
+
+_&part;u / &part;t = &alpha;&nabla; &sup2;u_
+
+Where:
+* _u_ = Temperature at a position, _x_
+* _t_ = time
+* _&alpha;_  = thermal diffusity of the material
+* _&nabla;_ = The LaPlace operator
+
+### FTCS: Assumptions and Discretization
+To make the problem more approachable, we make some simplifying assumptions:
+* _&alpha;_ is constant
+* The heat/cooling source only comes from initial/boundary conditions
+* Only Cartesian coordinates in 1 dimension are used
+
+Since the heat equation takes the form of a PDE over a continuous space, most programs create a discretization of the formula over a finite set of sampling points. By approximating the first derivative of _&part;u / &part;t_, the equation can be discretized to:
+
+_u_<sup>_k_+1</sup><sub>_i_</sub> = _ru_<sup>_k_</sup><sub>_i_+1</sub> + (1 - 2 _r_)_u_<sup>_k_</sup><sub>_i_</sub> + _ru_<sup>_k_</sup><sub>_i_-1</sub>
+(The full discretization process can be found [here](https://xsdk-project.github.io/ATPESC2018HandsOnLessons/lessons/hand_coded_heat/ "Hand-Coded Heat").)
+
+Where _r = &alpha;_ &Delta;_t&sup2; /_ &Delta;_x&sup2;_. This is the Forward-Time, Central-Space (FTCS) scheme which is considered numerically stable as long as _r_ is less than or equal to 0.5.
+
+### Why the Heat Equation?
+PDEs are widely used among the different sciences to model various phenomena such as electromagnetism, fluid mechanics, and heat transfers. While PDEs extend across multiple fields, they can still be quite tricky to work with because unlike ordinary differential equations, they are comprised of derivatives of more than just a single variable. A single example of a PDE is the Heat Equation, which is used calculate the distribution of heat on a region over time. While here we just focus on the 1-dimensional version of the Heat Equation, it can actually take a multitude of forms including the [Fourier](https://fenix.tecnico.ulisboa.pt/downloadFile/3779571609326/transp3.pdf "Fourier Equation"), [LaPlace](https://www.britannica.com/science/Laplaces-equation "LaPlace's Equation") (also known as steady-state), [2D](http://ramanujan.math.trinity.edu/rdaileda/teach/s12/m3357/lectures/lecture_3_6_short.pdf "2D Heat Equation"), and [3D](https://ocw.mit.edu/courses/mathematics/18-303-linear-partial-differential-equations-fall-2006/lecture-notes/pde3d.pdf "3D Heat Equation") heat equations.
+
+By taking a continuous PDE and turning it into a discrete formula, parallelism can be accomplished. For example, the equation for finding the tempurature at a given point in time in the 1-dimensional Heat Equation is:
```

Comment:
IMHO, parallelism isn't so much a result of discretization as it is dividing the work of the algorithm into independently computable tasks.

Reply:
I guess that it's possible to get parallelization without discretization... Would changing the wording to the following work?

"By taking a continuous PDE and turning it into a discrete formula, it can be more easily parallelized."

Reply:
No. Discretization and parallelization are wholly orthogonal concepts though it can be argued I guess that a given discretization may enhance or inhibit the work involved in parallelizing it.

I am not sure what you are after here wrt parallelization but the text seemed to imply that *discretization* is what makes parallelization possible. Yes, I guess in that without discretization we cannot parallelize. But the ability to parallelize is rarely (if ever IMHO) the rationale for a given choice in discretization. In fact, I am not sure I am aware of any discretization techniques that we cannot also (easily) parallelize.

Reply:
In this *particular* example, parallelization divides the space dimension, whereas discretization divides the time dimension.

Am I expressing the distinction correctly?

Reply:
No, not really. In the continuous mathematical equation, both time and space dimensions are *continuous*. To evaluate the PDE with a digital computer algorithm, we must discretize *all* dimensions meaning time and space. The discretization in space is perhaps a little easier to "see" but the discretization in time is equally important to "see". It manifests as the fact that we *evaluate* the PDEs only at specific moments in time...in this case those expressed by `k*delta-t` where `k` is some integer. If we were using a more sophisticated time-integration algorith, we might *vary* delta-t as we go and take bigger steps in time as the dynamics smooth out. But, we'd still be *discretizing* in time. This is part of the reason I ask the reader if s/he can "see" a *mesh* [here](https://xsdk-project.github.io/MathPackagesTraining/lessons/hand_coded_heat/)? Indeed, there *is* a mesh as there and its important to be aware of that fact even if it is subtle. It serves as a launching pad for getting into more interesting (e.g. non uniform) meshing and discretization. Does that make sense?

Reply:
I think we addressed this comment. We wanted to leave in the fact that the heat equation is often used to teach parallel programming, but I think we've decoupled that from the idea of discretization.

Reply:
Pardon me...I hadn't followed closely enough to understand what changed in the PR vs. responding to questions raised here in the conversation about the PR.

If you think this is resolved (my main concern was to not have a statement that suggests that discretization alone is sufficient to *accomplish* parallelism), then feel free to resolve.